File: frontend/src/app/dto/order.dto.ts
import { UserBasicDTO } from './userBasic.dto';
import { OrderItemDTO } from './orderItem.dto';

export interface OrderDTO {
  id: number;
  user: UserBasicDTO;
  totalWeight: number;
  totalPrice: number;
  orderDate: string | null; 
  items: OrderItemDTO[];
}

File: backend/src/main/java/es/codeurjc/quesosbartolome/dto/OrderDTO.java
package es.codeurjc.quesosbartolome.dto;

import java.time.LocalDateTime;
import java.util.List;

public record OrderDTO(
    Long id,
    UserBasicDTO user,
    Double totalWeight,
    Double totalPrice,
    LocalDateTime orderDate,
    List<OrderItemDTO> items
) {}